### PR TITLE
feat(elements|ino-fab): add slot for leading icon

### LINKED ISSUE #222: ino-fab: Replace ino-icon property with ino-icon slot
### Describe the solution you'd like
For the sake of consistency, I would like to propose to replace the `inoIcon` property of the the `ino-fab` component with a `ino-icon` slot. However, feel free to share your opinion.

#### AC:

- [x]  The `inoIcon` property of the `ino-fab` component has been removed
- [x] A new `ino-icon` slot has been added to the `ino-fab` component

### PR DESCRIPTION
**Pull request by P1NHE4D**
⚙️ **Changes:**
* marked `ino-icon` property as deprecated
* added new `ino-icon-leading` slot to the `ino-fab` component
* Updated the respective stories accordingly

🎟️ **Closes #222**

### CODE REVIEW

File: packages/elements/src/components/ino-fab/ino-fab.tsx [hunk 1]
```diff
@@ -122,14 +130,17 @@ export class Fab implements ComponentInterface {
...
     });
 
+    const iconSlotHasContent = hasSlotContent(this.el, 'ino-icon-leading');
+
     return (
       <Host id={this.uniqueHelperId}>
         <button class={classFab} disabled={this.inoDisabled}>
           <span class="material-icons mdc-fab__icon">
-            {this.inoIcon ? (
-              <ino-icon class="mdc-button__icon" ino-icon={this.inoIcon} />
```
Comment (P1NHE4D) [inline packages/elements/src/components/ino-fab/ino-fab.tsx:130]: The `mdc-button__icon` class has been removed intentionally as it only decreases the size of the icon by 1px, which is insignificant. It has no noticeable effect on the appearance of the `ino-fab` component.

Comment (janivo) [reply packages/elements/src/components/ino-fab/ino-fab.tsx:130]: good to know, thanks for the heads-up

File: packages/storybook/src/stories/ino-fab/ino-fab.stories.js [hunk 2]
```diff
@@ -18,74 +18,93 @@ export const DefaultUsage = () => /*html*/ `
...
+  "--ino-fab-background-color-hover",
+  "#5d60f7",
+  "Custom Properties"
+)};
             --ino-fab-background-color-active: ${text(
-              '--ino-fab-background-color-active',
-              '#0d10f3',
-              'Custom Properties'
-            )};
+  "--ino-fab-background-color-active",
```
Comment (janivo) [inline packages/storybook/src/stories/ino-fab/ino-fab.stories.js:36]: This seems to be formatted with the new eslint rules. We should stick to our prettier format until we resolved #283

File: packages/elements/src/components/ino-fab/ino-fab.tsx [hunk 3]
```diff
@@ -122,14 +130,17 @@ export class Fab implements ComponentInterface {
...
-            ) : (
-              <slot></slot>
+            { this.inoIcon && !iconSlotHasContent && (
+              <ino-icon ino-icon={this.inoIcon}/>
+            )}
+            { iconSlotHasContent && (
+              <slot name="ino-icon-leading"/>
```
Comment (janivo) [inline packages/elements/src/components/ino-fab/ino-fab.tsx:143]: Stencil introduced annotations for slots which will be used to generate a table in the readme. We should use it here as well.

See:
https://github.com/inovex/elements/blob/8b91f4e51abf5a920ccf3718ccc54e13802768f1/packages/elements/src/components/ino-button/ino-button.tsx#L16-L19

**Review by janivo — CHANGES_REQUESTED** [review]

File: packages/elements/src/components/ino-fab/ino-fab.tsx (hunk 3 above)
Comment (P1NHE4D) [reply packages/elements/src/components/ino-fab/ino-fab.tsx:143]: ✅

File: packages/storybook/src/stories/ino-fab/ino-fab.stories.js (hunk 2 above)
Comment (P1NHE4D) [reply packages/storybook/src/stories/ino-fab/ino-fab.stories.js]: whoops... should be fixed now

**Review by janivo — APPROVED** [review]

**Pull request merged**